### PR TITLE
[BUG] Docker distribution builds are failing. Switching to http://vault.centos.org

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Docker distribution builds are failing across the board

```
CentOS Linux 8 - AppStream                      144  B/s |  38  B     00:00    
| �[91mError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
| �[0myum error: retry 1 in 10s
| CentOS Linux 8 - AppStream                      176  B/s |  38  B     00:00    
| �[91mError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
| �[0myum error: retry 2 in 10s
| CentOS Linux 8 - AppStream                      228  B/s |  38  B     00:00    
| �[91mError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
| �[0myum error: retry 3 in 10s
| CentOS Linux 8 - AppStream                      209  B/s |  38  B     00:00    
| �[91mError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
| �[0myum error: retry 4 in 10s
| CentOS Linux 8 - AppStream                      405  B/s |  38  B     00:00    
| �[91mError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
| �[0myum error: retry 5 in 10s
| CentOS Linux 8 - AppStream                       86  B/s |  38  B     00:00    
| �[91mError: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
| �[0myum error: retry 6 in 10s

```
 
### Issues Resolved
Closes https://github.com/opensearch-project/OpenSearch/issues/2023
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
Thanks for the change @reta. I was looking into the same issue and found the below 
```
sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*

sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*
```
Let me know what you think?

**Comment by reta** [comment]
> Thanks for the change @reta. I was looking into the same issue and found the below
> 
> ```
> RUN sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*
> 
> RUN sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*
> ```
> 
> Let me know what you think?

Thanks @owaiskazi19, I haven't tried this way yet (seems like it may not be sufficient) but let me check it

**Comment by reta** [comment]
@owaiskazi19 I think your approach with changing the repo is better and does keep CentOS 8 distribution (instead of updating to Stream), thank you, I updated the pull request with this change.

**Comment by owaiskazi19** [comment]
> @owaiskazi19 I think your approach with changing the repo is better and does keep CentOS 8 distribution (instead of updating to Stream), thank you, I updated the pull request with this change.

I'm glad it worked out. Thanks for considering the suggestion. Keeping the reference here: https://www.centos.org/centos-linux-eol/

**Review by owaiskazi19 — APPROVED** [review]
Can you please edit the issue with the proposed solution? Thanks.

**Review by andrross — APPROVED** [review]

**Comment by owaiskazi19** [comment]
@reta @andrross do we need to backport this change to other branches like 1.1, 1.2.. as well?

File: distribution/docker/src/docker/Dockerfile [hunk 1]
```diff
@@ -63,6 +63,11 @@ FROM ${base_image}
 
 ENV OPENSEARCH_CONTAINER true
 
+RUN  dnf --disablerepo '*' --allowerasing -y \\
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply distribution/docker/src/docker/Dockerfile]: Seems like you have a better approach by changing the repo base url.
Thanks.

**Comment by reta** [comment]
> @reta @andrross do we need to backport this change to other branches like 1.1, 1.2.. as well?

Sorry, missed that, for 1.2 - definitely, for 1.1 - not sure

**Pull request merged**